ocs_ci/ocs/amq.py

```diff
@@ -569,11 +571,39 @@ def run_amq_benchmark(
         else:
             cmd = "bin/benchmark --drivers /driver_kafka /amq_workload.yaml"
         log.info(f"Run benchmark and running command {cmd} inside the benchmark pod ")
+
+        if run_in_bg:
+            executor = ThreadPoolExecutor(1)
+            result = executor.submit(
+                self.run_amq_workload_in_bg,
+                cmd, benchmark_pod_name, tiller_namespace, timeout
+            )
+            return result
+
         pod_obj = get_pod_obj(name=f"{benchmark_pod_name}-driver", namespace=tiller_namespace)
         result = pod_obj.exec_cmd_on_pod(command=cmd, out_yaml_format=False, timeout=timeout)
 
         return result
 
+    def run_amq_workload_in_bg(
+        self, command, benchmark_pod_name, tiller_namespace, timeout
+    ):
+        """
+        Runs amq workload in bg
+
+        Args:
+             command (str): Command to run on pod
+             benchmark_pod_name (str): Pod name
+             tiller_namespace (str): Namespace of pod
+             timeout (int): Time to complete the run
+
+        Returns:
+            Thread: A thread of the amq workload execution
```

Comment:
returning exec_cmd_on_pod which will not return thread. 
Am I missing anything?

Reply:
No, changed accordingly